### ISSUE TITLE
run_spp, not tag.lwcc error

### ISSUE THREAD

**Issue by KunFang93** [kundajelab/phantompeakqualtools#13]
Hi, 

I have a problem when I run run_spp.R:
Error in argfun(job) : object 'tag.lwcc' not found
Calls: find.binding.positions ... sendData -> sendData.SOCKnode -> serialize -> argfun
Execution halted
Thanks a lot!

Kun

**Comment by leepc12** [comment]
Please post your full command line and any helpful error logs.

**Comment by KunFang93** [comment]
my full command is:
run_spp.R -p=20 -c=ENCFF000ZLI.filt.nodup.srt.SE.tagAlign.gz -i=ENCFF162ZAB.filt.nodup.srt.SE.tagAlign.gz -npeak=300000 -odir=test2 -speak=205 -savr -savp -rf -out=test2/test.ccscores

log:
ChIP data: ENCFF000ZLI.filt.nodup.srt.SE.tagAlign.gz 
Control data: ENCFF162ZAB.filt.nodup.srt.SE.tagAlign.gz 
strandshift(min): -500 
strandshift(step): 5 
strandshift(max) 1500 
user-defined peak shift 205 
exclusion(min): 10 
exclusion(max): NaN 
num parallel nodes: 20 
FDR threshold: 0.01 
NumPeaks Threshold: 3e+05 
Output Directory: test2 
narrowPeak output file name: NA 
regionPeak output file name: test2/ENCFF000ZLI.filt.nodup.srt.SE.tagAlign_VS_ENCFF162ZAB.filt.nodup.srt.SE.tagAlign.regionPeak 
Rdata filename: NA 
plot pdf filename: test2/ENCFF000ZLI.filt.nodup.srt.SE.tagAlign.pdf 
result filename: test2/test.ccscores 
Overwrite files?: TRUE

Decompressing ChIP file
Decompressing control file
Loading required package: Rcpp
Reading ChIP tagAlign/BAM file ENCFF000ZLI.filt.nodup.srt.SE.tagAlign.gz 
opened /tmp/Rtmpvb55A1/ENCFF000ZLI.filt.nodup.srt.SE.tagAlign6cc93f9df68a
done. read 18492897 fragments
ChIP data read length 32 
[1] TRUE
Reading Control tagAlign/BAM file ENCFF162ZAB.filt.nodup.srt.SE.tagAlign.gz 
opened /tmp/Rtmpvb55A1/ENCFF162ZAB.filt.nodup.srt.SE.tagAlign6cc94cf5b177
done. read 10043693 fragments
Control data read length 32 
Calculating peak characteristics
Minimum cross-correlation value 0.2042988 
Minimum cross-correlation shift 1500 
Top 3 cross-correlation values 0.220347194048325 
Top 3 estimates for fragment length 205 
Window half size 505 
Phantom peak location 30 
Phantom peak Correlation 0.21713 
Normalized Strand cross-correlation coefficient (NSC) 1.078554 
Relative Strand cross-correlation Coefficient (RSC) 1.250731 
Phantom Peak Quality Tag 1 
null device 
          1 
Removing read stacks
Finding peaks
finding background exclusion regions ... done
determining peaks on provided 1 control datasets:
using reversed signal for FDR calculations
bg.weight= 0.543123  Error in argfun(job) : object 'tag.lwcc' not found
Calls: find.binding.positions ... sendData -> sendData.SOCKnode -> serialize -> argfun
Execution halted

Thanks!

**Comment by leepc12** [comment]
Did you install all R requirements including `spp` package?

**Comment by KunFang93** [comment]
I used conda install -c bioconda phantompeakqualtools to install the software. It installed the dependence r-spp.

**Comment by leepc12** [comment]
I think you also have ENCODE pipeline's conda environment. Can you try with that?

```
$ source activate encode-atac-seq-pipeline
$ Rscript run_spp.R ...
```

**Comment by KunFang93** [comment]
Unfortunately, I don't have the pipeline now....I run tutorial successfully once but when I try to run tutorial again. It failed again. And after I reinstall couple of times, my conda environment even doesn't use its own R but use the base environment's R. I completely screwed up my old conda environment so I reinstall a new miniconda3 and decided manually follow the pipeline in the google doc....
By the way, before this error, there is another error when I run the run_spp.R: no function runmean, but this can be solved by add library(caTools) before line 718: cc$y <-runmean(cc$y,sbw,alg="fast").

So sorry for bothering you a lot, thanks!

**Comment by leepc12** [comment]
I got the same `tag.lwcc` error with R 3.6 with spp 1.16.
I downgraded them to R 3.4.4 with spp 1.15 worked fine for me.

Until we fix this issue, downgrade your R  to <=3.4.4 and R spp package to <=1.15.

**Comment by KunFang93** [comment]
Hi, Lee

Thanks for your reply~ I will do that

Best,
Kun

**Comment by leepc12** [comment]
Fixed in 1.2.2. Closing this.

**Comment by annashcherbina** [comment]
Hi Jin, I'm still seeing this issue. what do you mean by "fixed in 1.2.2?" 1.2.2 of phantompeaktools? 

ChIP data: mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted.bam 
Control data: mouse_chipseq-Input_G1E_307C.2x36mers.mm10.unique.dedup.bam 
strandshift(min): -100 
strandshift(step): 5 
strandshift(max) 600 
user-defined peak shift NA 
exclusion(min): 10 
exclusion(max): NaN 
num parallel nodes: 20 
FDR threshold: 0.01 
NumPeaks Threshold: 3e+05 
Output Directory: SPP-300K-mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted 
narrowPeak output file name: NA 
regionPeak output file name: SPP-300K-mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted/mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted_VS_mouse_chipseq-Input_G1E_307C.2x36mers.mm10.unique.dedup.regionPeak 
Rdata filename: NA 
plot pdf filename: SPP-300K-mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted/mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted.pdf 
result filename: NA 
Overwrite files?: TRUE

[1] TRUE
[1] TRUE
Loading required package: Rcpp
Reading ChIP tagAlign/BAM file mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted.bam 
opened /tmp/RtmpVejf8I/mouse_chipseq-ChIP_G1E_307C.2x36mers.mm10.unique.dedup.merged.sorted.tagAlign46c112d661c
done. read 64702726 fragments
ChIP data read length 36 
[1] TRUE
Reading Control tagAlign/BAM file mouse_chipseq-Input_G1E_307C.2x36mers.mm10.unique.dedup.bam 
opened /tmp/RtmpVejf8I/mouse_chipseq-Input_G1E_307C.2x36mers.mm10.unique.dedup.tagAlign46c459054c6
done. read 67907542 fragments
Control data read length 36 
Calculating peak characteristics
Minimum cross-correlation value 0.4602906 
Minimum cross-correlation shift -100 
Peak cross-correlation value 0.657640058078151 
Peak strand shift 235 
Window half size 470 
Phantom peak location 40 
Phantom peak Correlation 0.5083081 
Normalized cross-correlation coefficient (NCCC) 1.42875 
Relative Cross correlation Coefficient (RCCC) 4.109946 
Phantom Peak Quality Tag 2 
null device 
          1 
Removing read stacks
Finding peaks
finding background exclusion regions ... done
determining peaks on provided 1 control datasets:
using reversed signal for FDR calculations
bg.weight= 2.039481  Error in argfun(job) : object 'tag.lwcc' not found
Calls: find.binding.positions ... sendData -> sendData.SOCKnode -> serialize -> argfun
Execution halted

**Comment by annashcherbina** [comment]
sorry, I believe I had the wrong tag. seems fine in 1.2.2. Closing again.
